### PR TITLE
[SILOptimizer] Eliminate dead alloc_pack

### PR DESCRIPTION
**Pull request by Azoy**
This is very rudimentary support for eliminating dead packs.

### CODE REVIEW

File: test/SILOptimizer/dead_alloc_pack.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// CHECK-NEXT: return [[RET]] : $()
+public func makeTrivialTuple() {
+  makeTuple(123, 321, 456, 654)
+}
+
+// CHECK-LABEL: sil @$s3dap26makeTrivialDestructorTupleyyF : $@convention(thin) () -> () {
+// CHECK: bb0:
+// CHECK-NEXT: %0 = alloc_pack $Pack{TrivialDestructor, TrivialDestructor}
+public func makeTrivialDestructorTuple() {
+  makeTuple(TrivialDestructor(), TrivialDestructor())
```
Comment (Azoy) [inline test/SILOptimizer/dead_alloc_pack.swift:50]: Is this also a case that should be eliminated? It seems we make 2 packs to pass to `makeTuple`, but we also consider this pack to be non-trivial.. is that correct?

**Comment by Azoy** [comment]
@swift-ci please test

**Review by eeckstein — COMMENTED** [review]
Basically LGTM.
Can you also add a SIL test? E.g. in `test/SILOptimizer/dead_alloc_elim.sil`

**Comment by Azoy** [comment]
@swift-ci please test

**Review by eeckstein — APPROVED** [review]
lgtm!

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please clean test

**Pull request closed without merging**